app/controllers/api/providers_controller.rb

```diff
@@ -85,6 +85,21 @@ def import_vm_resource(type, id = nil, data = {})
       end
     end
 
+    def options
+      if params[:ems_type].blank?
+        all_options = Hash[
+          ExtManagementSystem.descendants.collect do |ems|
+            [ems.ems_type, ems.respond_to?(:provider_settings) ? ems.provider_settings : {}]
```

Comment:
Yes, those options themselves are per provider instance.
What I am serving here are only their descriptions but belong there as they are static and not configurable.